pkg/generate/generator.go

```diff
@@ -223,6 +223,9 @@ func (g *Generator) GetCRDs() ([]*ackmodel.CRD, error) {
 			}
 		}
 
+		crd.SpecIdentifierField = crd.GetSpecIdentifierField()
+		crd.IsAdoptable = g.cfg.ResourceIsAdoptable(crdName)
```

Comment:
For the above, please move the assignment of CRD fields into the `ackmodel.NewCRD` function.